### PR TITLE
Separate function to arrow transform

### PR DESCRIPTION
**Pull request by bouk**
@lemonmade

### CODE REVIEW

File: packages/shopify-codemod/test/transforms/function-to-arrow.test.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import ternaryStatementToIfStatement from 'function-to-arrow';
+
+describe('functionToArrow', () => {
+  it('transforms the simple case', () => {
+    expect(ternaryStatementToIfStatement).to.transform('function-to-arrow/basic');
+  });
+
+  it("doesn't transform function that use `this`", () => {
+    expect(ternaryStatementToIfStatement).to.transform('function-to-arrow/this');
+  });
```
Comment (lemonmade) [inline packages/shopify-codemod/test/transforms/function-to-arrow.test.js]: Can you add a function that it still converts a function that has a nested `this` in a callback? Something like:

```js
const foo = function() {
  bar(function() {
    this.baz;
  });
}
```

should become:

```js
const foo = () => {
  bar(function() {
    this.baz;
  });
}
```

Comment (lemonmade) [inline packages/shopify-codemod/test/transforms/function-to-arrow.test.js]: Although this will probably make the mod a fair bit more complex.

Comment (bouk) [inline packages/shopify-codemod/test/transforms/function-to-arrow.test.js]: that specific example actually gets turned into

```
function foo() {
  bar(function() {
    this.baz;
  });
}
``

:smile:

Comment (bouk) [inline packages/shopify-codemod/test/transforms/function-to-arrow.test.js]: but I don't think we should support that specific case, probably doesn't happen too often and it's more trouble than it's worth

Comment (lemonmade) [inline packages/shopify-codemod/test/transforms/function-to-arrow.test.js]: :+1:

Comment (lemonmade) [inline packages/shopify-codemod/test/transforms/function-to-arrow.test.js]: Add docs!

**Pull request merged**